src/runtime_src/hip/core/stream.cpp

```diff
@@ -27,7 +27,7 @@ stream::
 
 void
 stream::
-enqueue(std::shared_ptr<command>&& cmd)
+enqueue(std::shared_ptr<command> cmd)
```

Comment:
Yes, by-value here makes sense since you move in m_cmd_queue, but up the chain it should be `const std::shared_ptr<command>& cmd` to avoid reference count logic.

Incidentally, the std::move on line 51 below should be removed.